### PR TITLE
Add Kyuubi Hive JDBC Module

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Aiming to make a better supported client for Kyuubi and Spark

- Add catalog to getTables meta function for DataLakes (DONE)
- Deploy to maven central (PENDING)

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/pom.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    <artifactId>kyuubi-hive-jdbc</artifactId>
+    <name>Kyuubi Project Hive JDBC Client</name>
+    <packaging>jar</packaging>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.apache.kyuubi</groupId>
+            <artifactId>kyuubi-common</artifactId>
```
Comment (pan3793) [inline kyuubi-hive-jdbc/pom.xml:37]: If it depends on kyuubi-common, it's also depends on scala.

File: kyuubi-hive-jdbc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        <dependency>
+            <groupId>org.apache.hive</groupId>
+            <artifactId>hive-serde</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
+            <version>2.6</version>
```
Comment (pan3793) [inline kyuubi-hive-jdbc/pom.xml:69]: extract version to root `pom.xml` `<dependencyManagement>`

File: kyuubi-hive-jdbc/pom.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        </dependency>
+        <dependency>
+            <groupId>commons-collections</groupId>
+            <artifactId>commons-collections</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.8.0</version>
```
Comment (pan3793) [inline kyuubi-hive-jdbc/pom.xml:111]: ditto

File: pom.xml [hunk 4]
```diff
@@ -289,12 +292,25 @@
...
+                <version>${commons-lang.version}</version>
+            </dependency>
+
             <dependency>
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-lang3</artifactId>
                 <version>${commons-lang3.version}</version>
             </dependency>
 
+
```
Comment (pan3793) [inline pom.xml:313]: unnecessary new line

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, will merge it, and open #765 as a follow-up.
